### PR TITLE
[error] [doc] Better error message when X11 environment failed to initialize

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1817

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

**Review by thinking-tower — APPROVED** [review]
LGTM! Thanks for moving the fps stuff!

**Comment by archibate** [comment]
> All required statuses and check runs on this pull request must run successfully to enable automatic merging. 

@yuanming-hu Can we use an `or` parse in this 'required statuses` mechanism? I.e.

`(Build and Test && Code format) || Checks the Workflow Run of the Previous Commit`

LHS for non-gardener final-commit, RS for gardener final-commit, which always fix the code format.

File: docs/gui.rst [hunk 1]
```diff
@@ -10,22 +10,36 @@ Create a window
...
     The following code creates a window of resolution ``640x360``:
 
-    ::
+    .. code-block:: python
 
         gui = ti.GUI('Window Title', (640, 360))
 
+    .. note::
+
+        If you're running Taichi on a machine without GUI environment, consider
```
Comment (yuanming-hu) [inline docs/gui.rst:32]:
```suggestion
        If you are running Taichi on a machine without a GUI environment, consider
```

**Review by yuanming-hu — APPROVED** [review]
Thanks! LGTM. Just one wording nit.

> @yuanming-hu Can we use an `or` parse in this 'required statuses` mechanism? I.e.
> 
> `(Build and Test && Code format) || Checks the Workflow Run of the Previous Commit`
> 
> LHS for non-gardener final-commit, RS for gardener final-commit, which always fix the code format.

GitHub doesn't provide that option, unfortunately...

**Pull request merged**